File: jasmine/lib/jasmine/currency.ex
defmodule Jasmine.Currency do
  use Ecto.Schema
  import Ecto.Changeset

  schema "currencies" do
    field :code, :string
    field :number, :string
    field :name, :string
    field :country, :string
    field :units, :integer
    field :created_at, :utc_datetime_usec
  end

  @doc false
  def changeset(currency, attrs) do
    currency
    |> cast(attrs, [:code, :number, :name, :country, :units, :created_at])
    |> validate_required([:code, :number, :name, :country, :units])
  end
end

File: dahlia/src/main/kotlin/com/github/saturn_siv/palm/dahlia/plugins/cms/models/Currency.kt
package com.github.saturn_siv.palm.dahlia.plugins.cms.models

import jakarta.persistence.*
import java.time.Instant

@Table(name = "currencies")
@Entity(name = "currency")
class Currency(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    var id: Int,
    @Column(nullable = false)
    var code: String,
    @Column(nullable = false)
    var number: String,
    @Column(nullable = false)
    var name: String,
    @Column(nullable = false)
    var country: String,
    @Column(nullable = false)
    var units: Int,
    @Column(nullable = false)
    var createdAt: Instant,
)